implementations/rust/ockam/ockam_api/src/ui/command.rs

```diff
@@ -0,0 +1,101 @@
+use std::fmt::{Debug, Formatter};
+
+#[derive(Clone, Debug)]
+pub struct Commands {
+    pub(crate) commands: Vec<Command>,
+}
+
+#[derive(Clone)]
+pub(crate) struct Command {
+    pub(crate) name: String,
+    pub(crate) custom_name: String,
+}
+
+impl Debug for Command {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Command")
+            .field("name", &self.name)
+            .field("custom_name", &self.custom_name)
+            .finish()
+    }
+}
+
+impl Commands {
+    pub fn new(commands: &str) -> Self {
+        let commands = commands
+            .split(',')
+            .filter_map(|c| {
+                if c.is_empty() {
+                    return None;
+                }
+                let mut parts = c.split('=');
+                let name = match parts.next() {
+                    Some(name) => name,
+                    None => return None,
+                };
+                let custom_name = parts.next().unwrap_or(name);
+                Some(Command {
+                    name: name.to_string(),
+                    custom_name: custom_name.to_string(),
+                })
+            })
+            .collect();
+        Self { commands }
+    }
+
+    pub fn hide(&self, command_name: &'static str) -> bool {
+        if self.commands.is_empty() {
+            return false;
+        }
+        // we only support custom names for top-level commands
```

Comment:
It might be a good idea to give an example here of what is a "top-level" command and what it's not.
The fact that we are identifying this with the presence of a whitespace seems a bit fragile to me.

Reply:
> It might be a good idea to give an example here of what is a "top-level" command and what it's not.
The fact that we are identifying this with the presence of a whitespace seems a bit fragile to me.

I thought it was a limitation in clap, but it turns out that you can name commands with spaces. I'm working on a different approach.

Reply:
I moved that check to the brand.rs binary. Otherwise, we end up in a circular dependency situation when trying to parse the command and get the top-level commands.

https://github.com/build-trust/ockam/compare/86ad15c37431ed5e2ee94699e6ffb14127f3aea2..371f8db10659c81c4d4738495141f616161e6fe8